docs/adr/2022-04-26__alerting.md

```diff
@@ -0,0 +1,295 @@
+# Alerting in OpMon
+
+* Author: Anna Scholtz
+* Date: 2022-04-26
+* Status: Accepting Feedback (Feedback from DS is especially welcome)
+
+## Introduction
+
+Users of OpMon want to be able to see critical changes and optionally get notified whenever they occur. This ADR describes how OpMon will provide the ability to define different monitors, how they will be implemented and how users can subscribe to getting email notifications.
+
+As an example, of what the operational monitoring dashboards currently look like see: https://mozilla.cloud.looker.com/dashboards/operational_monitoring::bug_1715474_rollout_yandex_sponsored_tile_rollout_release_89_100?Percentile=50
+The dashboards have a tile for each metric which shows the values of all available branches for the percentile currently selected as filter.
+
+## Constraints
+
+There are a couple of constraints when checking for critical changes in OpMon:
+* **Cost:** We will want to keep the cost low since there will be a lot of dashboards (for example, we create one for each rollout)
+* **Computation Speed:** The operational monitoring dashboards get updated on a daily basis and there are usually dozens of dashboards that need to get updated. Ideally, the checks for each dashboard and day finish within minutes rather than hours.
+* **Minimize False Positives:** Sending alerts when nothing is wrong too many times results in users simply ignoring alerts at some point
+
+## Monitor Types
+
+When talking to users there are a few different cases when they would like to receive an alert:
+* **Large differences between branches:** Whenever the confidence intervals of different branches for a specific metric no longer overlap, it indicates that there is potentially some significant difference.
+* **Thresholds:** Comparing the values of a metric to a user-defined threshold.
+* **Deviation from historical data:** Detect anomalous behaviour of a metric based on previously collected data.
+
+### Large differences between branches
+
+The OpMon dashboards show the values for specific metrics as a line chart with confidence intervals. Each line represents the metric values for a different branch. Whenever the confidence intervals of the branches do not overlap, it is considered a critical change. See:
+
+<img src="./images/alerting_branch_differences.png" width="500"/>
+
+These checks need to be manually configured for specific metrics/probes. The OpMon configuration files will be extended to something like this:
+
+```toml
+[alert]
+
+[alert.crashes] # new alert
+type = "ci"
+probes = [      # list of metrics to check
+    "startup_crashes", 
+    "main_crashes", 
+    "oom_crashes"
+]
+percentiles = [50, 90]  # percentiles that should be considered
+```
+
+
+### Thresholds
+
+In some cases the expected value of a metric is known and any large deviation from that expected value is considered a critical change. Fixed thresholds can be used to specify when a value is too large or too low. See:
```

Comment:
What I said above applies to comparing two CIs to each other. If you're comparing a CI against a known value, if the CI excludes the value then it's statistically significant, which is what we'd expect.